generate_table.Rmd

```diff
@@ -0,0 +1,624 @@
+                          "//Return//ReturnHeader//ReturnTypeCd",
+                          "//Return//ReturnHeader//TaxPeriodBeginDt",
+                          "//Return//ReturnHeader//TaxPeriodEndDt",
+                          "//AmendedReturnInd",
+                          "//Return//ReturnHeader//BusinessName")
+
+
+
+
+#' @param variables character vector of variables to extract from the xml files
+#' @param filename string containing full path to xml file to be read
+#' @param include_paths logical; whether to include paths to extracted variables in data frame
+
+get_df <- function(variables,
+                   filename, 
+                   include_paths = TRUE) {
+  
+  # create column names with just the variables (no paths)
+  variables_no_path <- gsub("*.*\\/", "", variables )
+  
+  xml_file <- read_xml(filename)
+  xml_file <- xml_ns_strip(xml_file)
+  
+  # extract each variable; if it isn't present, put NA 
+  extracted <- map(variables, ~{
+    value <- xml_find_all(
+      xml_file, 
+      xpath =.x)
+    value <- ifelse(length(value) ==0, 
+                    NA, 
+                    xml_text(value)) })
+  
+   names(extracted) <- variables_no_path
+   
+   extracted <- extracted %>%
+     as_tibble()
+ 
+ 
+   # add columns with the paths to each variable within the xml file
+   if(include_paths) {
+      paths <- map(variables, ~ {
+        value <- xml_find_all(
+          xml_file, 
+          xpath = paste0(.x)) 
+         value <- ifelse(length(value) ==0, 
+                      NA, 
+                      xml_path(value)) })
+      
+      names(paths) <- paste0(variables_no_path, "_path")
+      
+      extracted <- extracted %>%
+        bind_cols(as_tibble(paths)) 
+  
+   }
+    
+   # check how many of the entries are NA
+   # if all NA, prefix 'irs:' may be needed
+   columns_not_na <- map_dbl(as.data.frame(extracted), ~!is.na(.x)) %>% sum()
+   
+   # handle case where prefix 'irs' is in front
+   if(columns_not_na == 0){
+     
+       extracted <- map(variables_no_path, ~ {
+        value <- xml_find_all(xml_file, 
+                              xpath = paste0("//irs:", .x)) 
+        value <- ifelse(length(value) ==0, 
+                    NA, 
+                    xml_text(value)) })
+        names(extracted) <- variables_no_path
+        
+         extracted <- extracted %>%
+           as_tibble()
+     
+        if(include_paths) {
+        paths <- map(variables_no_path, ~ {
+          value <- xml_find_all(
+            xml_file, 
+            xpath = paste0("//irs:", .x))
+      
+          value <- ifelse(length(value) ==0, 
+                      NA, 
+                      xml_path(value)) })
+        
+        names(paths) <- paste0(variables_no_path, "_path")  
+        
+      extracted <-  extracted %>%
+        bind_cols(as_tibble(paths)) 
+     }
+   }
+   
+ # add name of file to data frame
+ extracted %>%
+   mutate(filename = gsub("*.*\\/", "", filename))
+}
+
+
+
```

Comment:
Can you add a comment or two more here to indicate what it's doing?  From what I can tell it is: 
- Creating a new folder ./data
- Applying get_df to all files with the earlier specified variables to extract
- Wrangling all-data to have proper formatting and new variables like fiscal_year
- Saving the wrangled table for future use